### PR TITLE
feat(module-analyzer): Enable registering ViewEngineHooks by convention.

### PR DESCRIPTION
**Pull request by davismj**
depends on https://github.com/aurelia/binding/pull/423.

### CODE REVIEW

File: src/module-analyzer.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import {Container} from 'aurelia-dependency-injection';
 import {TemplateRegistryEntry} from 'aurelia-loader';
 import {ValueConverterResource} from 'aurelia-binding';
 import {BindingBehaviorResource} from 'aurelia-binding';
+import {ViewEngineHooksResource} from 'aurelia-binding';
```
Comment (alvarezmario) [inline src/module-analyzer.js]: @davismj now that you are at it, why not make just one import from aurelia-binding?

File: src/module-analyzer.js [hunk 2]
```diff
@@ -274,6 +275,9 @@ export class ModuleAnalyzer {
         } else if (conventional = BindingBehaviorResource.convention(key)) {
           resources.push(new ResourceDescription(key, exportedValue, conventional));
           metadata.define(metadata.resource, conventional, exportedValue);
+        } else if (conventional = ViewEngineHooksResource.convention(key)) {
```
Comment (alvarezmario) [inline src/module-analyzer.js]: @davismj the following 2 lines are exactly as the above, so why not move this to the previous `else-if`?

Comment (alvarezmario) [inline src/module-analyzer.js]: @davismj In fact, I think you can do a little refactor here starting on https://github.com/davismj/templating/blob/6cf33d84b7ded2d5b3cefcb9bc79b3f02c42e3d8/src/module-analyzer.js#L272. 

```
else if (conventional = 
  ValueConverterResource.convention(key) 
  || BindingBehaviorResource.convention(key) 
  || ViewEngineHooksResource.convention(key)
) {
```
this looks like a nice refactor, don't you agree?

File: src/module-analyzer.js (hunk 1 above)
Comment (davismj) [inline src/module-analyzer.js]: I thought about it, but I don't do a lot of contributing and I wanted to keep with the current conventions / style.

File: src/module-analyzer.js (hunk 2 above)
Comment (davismj) [inline src/module-analyzer.js]: Sort of the same reason as above. This also applies to the existing code for `ValueConverterResource` and `BindingBehaviorResource`, so I just followed the convention.

**Comment by davismj** [comment]
Nice suggestions @alvarezmario, I'll update my PR.

File: src/module-analyzer.js (hunk 1 above)
Comment (alvarezmario) [inline src/module-analyzer.js]: Oh, I see, it makes sense, but they probably will like it :smile: ping @EisenbergEffect

Comment (EisenbergEffect) [inline src/module-analyzer.js]: Feel free to collapse those into a single import. It's just an accidental oversight in this case.

**Pull request closed without merging**